src/app/custom-navigation-mode/custom-navigation-mode.module.ts

```diff
@@ -0,0 +1,19 @@
+import {NgModule} from '@angular/core';
+import {CommonModule} from '@angular/common';
+import {ArchwizardModule, ArchwizardModuleConfig} from 'angular-archwizard';
+import {CustomNavigationModeComponent} from './custom-navigation-mode.component';
+import {CustomNavigationModeFactory} from './custom-navigation-mode-factory.provider';
+
+const config: ArchwizardModuleConfig = {
+  navigationModeFactory: CustomNavigationModeFactory,
+};
+
+@NgModule({
+  imports: [
+    CommonModule,
+    ArchwizardModule.forRoot(config)
```

Comment:
Maybe also add a comment here why `forRoot` is used.

Reply:
Also done in 22f2f6e.